pkg/frontend/querymiddleware/codec.go

```diff
@@ -270,16 +273,17 @@ func (prometheusCodec) decodeRangeQueryRequest(r *http.Request) (MetricsQueryReq
 		return nil, err
 	}
 
-	result.Query = r.FormValue("query")
```

Comment:
any use of formvalue is a potential bug as it consumes the body on a POST.
everything should use `util.ParseRequestFormWithoutConsumingBody`, then work with the map of values

Reply:
Should we have a linting rule for this?